src/Dapr.Actors.Generators/ActorClientGenerator.cs

```diff
@@ -20,284 +26,265 @@ namespace Dapr.Actors.Generators;
+    /// <inheritdoc />
+    public void Initialize(IncrementalGeneratorInitializationContext context)
     {
-        private readonly List<INamedTypeSymbol> models = new();
-
-        public IEnumerable<INamedTypeSymbol> Models => this.models;
-
-        #region ISyntaxContextReceiver Members
-
-        public void OnVisitSyntaxNode(GeneratorSyntaxContext context)
+        // Register the source output that generates the attribute definitions for ActorMethodAttribute and GenerateActorClientAttribute.
+        context.RegisterPostInitializationOutput(context =>
         {
-            if (context.Node is not InterfaceDeclarationSyntax interfaceDeclarationSyntax
-                || interfaceDeclarationSyntax.AttributeLists.Count == 0)
-            {
-                return;
-            }
-
-            var interfaceSymbol = context.SemanticModel.GetDeclaredSymbol(interfaceDeclarationSyntax) as INamedTypeSymbol;
-
-            if (interfaceSymbol is null
-                || !interfaceSymbol.GetAttributes().Any(a => a.AttributeClass?.ToString() == GenerateActorClientAttributeFullTypeName))
-            {
-                return;
-            }
-
-            this.models.Add(interfaceSymbol);
-        }
-
-        #endregion
+            context.AddSource(
+                $"{Constants.ActorMethodAttributeFullTypeName}.g.cs",
+                Templates.ActorMethodAttributeSourceText(Constants.GeneratorsNamespace));
+
+            context.AddSource(
+                $"{Constants.GenerateActorClientAttributeFullTypeName}.g.cs",
+                Templates.GenerateActorClientAttributeSourceText(Constants.GeneratorsNamespace));
+        });
+
+        // Register the value provider that triggers the generation of actor clients when detecting the GenerateActorClientAttribute.
+        IncrementalValuesProvider<ActorClientDescriptor> actorClientsToGenerate = context.SyntaxProvider
+            .ForAttributeWithMetadataName(
+                Constants.GenerateActorClientAttributeFullTypeName,
+                predicate: static (_, _) => true,
+                transform: static (gasc, cancellationToken) => CreateActorClientDescriptor(gasc, cancellationToken));
+
+        // Register the source output that generates the actor clients.
+        context.RegisterSourceOutput(actorClientsToGenerate, GenerateActorClientCode);
     }
 
-    #region ISourceGenerator Members
-
-    /// <inheritdoc />
-    public void Execute(GeneratorExecutionContext context)
+    /// <summary>
+    /// Returns the descriptor for the actor client to generate.
+    /// </summary>
+    /// <param name="context">Current generator syntax context passed from generator pipeline.</param>
+    /// <param name="cancellationToken">Cancellation token used to interrupt the generation.</param>
+    /// <returns>Returns the descriptor of actor client to generate.</returns>
+    static ActorClientDescriptor CreateActorClientDescriptor(

```

Comment:
I presume the omission of an access modifier to this method is deliberate? Could we get away with this being private, or does it need to be internal (C# default)?

Same question for the `GenerateActorClientCode` method below.

Reply:
Fixed